### PR TITLE
Feature/make issue number hyperlinked

### PR DESCRIPTION
**Pull request by dxli94**
If there is any ticket number in the description, then should be hyperlinked. Check JIRA issue POET-44 QA statement for how to test.

### CODE REVIEW

**Review by haolinj — APPROVED** [review]
Looks good to me.

File: server/app/services/Utils.java [hunk 1]
```diff
@@ -1,26 +1,42 @@
...
-        (new FileWriter("logs/missedQuery.log", true));
-      out.write(missedQuestion+"\n");
-      out.close();
-    } catch (FileNotFoundException e) {
-      e.printStackTrace();
-    } catch (IOException e) {
-      e.printStackTrace();
-    }
+  public static String hyperlinkTicketNo(String issueDescription){
+    /**
```
Comment (dspasojevic) [inline server/app/services/Utils.java:16]: Please move this above the method that it is documenting.

File: server/app/services/languageProcessor/TaskMap.java [hunk 2]
```diff
@@ -54,7 +56,11 @@ public JsonNode getTicketDescription(String issueKey, JsonNode responseBody) {
     if (Extractor.extractString(responseBody, "description").equals("[\"Issue Does Not Exist\"]")) {
       return parseToJson("fail", configuration.getString("error-message.issue-not-found"));
     } else {
-      String answer = "Description of " + issueKey + " is as follows: \n" +
+      String IssueId = responseBody.get("key").toString().replaceAll("\"", "");
```
Comment (dspasojevic) [inline server/app/services/languageProcessor/TaskMap.java:59]: Please use camel case with a lower case first letter for variables. These are not constants, but constants should be all upper case with '_' between words, e.g. `SOME_CONSTANT_NAME`.

File: server/app/services/languageProcessor/TaskMap.java [hunk 3]
```diff
@@ -54,7 +56,11 @@ public JsonNode getTicketDescription(String issueKey, JsonNode responseBody) {
...
+      String IssueUrl = "http://jira.agiledigital.com.au/browse/" + IssueId;
+      String Hyperlink = "<" + IssueUrl+ "|" + IssueId + ">";
```
Comment (dspasojevic) [inline server/app/services/languageProcessor/TaskMap.java:61]: This code is the same as the code on line 80. This should be extracted out to a method, e.g. follow DRY principle.

File: server/app/services/languageProcessor/TaskMap.java [hunk 4]
```diff
@@ -54,7 +56,11 @@ public JsonNode getTicketDescription(String issueKey, JsonNode responseBody) {
     if (Extractor.extractString(responseBody, "description").equals("[\"Issue Does Not Exist\"]")) {
       return parseToJson("fail", configuration.getString("error-message.issue-not-found"));
     } else {
-      String answer = "Description of " + issueKey + " is as follows: \n" +
+      String IssueId = responseBody.get("key").toString().replaceAll("\"", "");
+      String IssueUrl = "http://jira.agiledigital.com.au/browse/" + IssueId;
```
Comment (dspasojevic) [inline server/app/services/languageProcessor/TaskMap.java:60]: This should be configured.

File: server/app/services/queryHandler/Extractor.java [hunk 5]
```diff
@@ -26,12 +27,9 @@ public static String extractString(JsonNode json, String key) {
     } else {
 
       if (key == "description") {
-        System.out.println("Check point 1");
-        System.out.println(json.get("fields").get(key).toString());
-        return json.get("fields").get(key).textValue();
+        String description = json.get("fields").get(key).textValue();
+        return Utils.hyperlinkTicketNo(description);
```
Comment (dspasojevic) [inline server/app/services/queryHandler/Extractor.java:31]: In Java, it is normal practice to have a single return from a method, e.g. assign what you are returning here and on line 33 to a variable and then return it at the end.

**Pull request merged**